### PR TITLE
add hashtag and consolidate libs

### PR DESCRIPTION
**Pull request by ryanzhang-oss**
1. combine two controller utils libs.
2. add the hash value to the app status
3. setup the render/apply with proper tags

### CODE REVIEW

File: pkg/appfile/parser.go [hunk 1]
```diff
@@ -22,7 +22,7 @@ const (
 	AppfileBuiltinConfig = "config"
 
 	// OAMApplicationLabel is application's metadata label tagged on AC and Component
```
Comment (wonderflow) [inline pkg/appfile/parser.go:24]: please also delete the annotation

File: pkg/controller/core.oam.dev/v1alpha2/application/apply_test.go [hunk 2]
```diff
@@ -91,7 +91,7 @@ var _ = Describe("Test Application apply", func() {
 		Expect(k8sClient.Delete(context.TODO(), &ns)).Should(Succeed())
 	})
 
-	It("Test creating applicationConfiguration revision", func() {
+	FIt("Test creating applicationConfiguration revision", func() {
```
Comment (wonderflow) [inline pkg/controller/core.oam.dev/v1alpha2/application/apply_test.go:94]:
```suggestion
	It("Test creating applicationConfiguration revision", func() {
```

File: pkg/controller/core.oam.dev/v1alpha2/applicationconfiguration/render.go [hunk 3]
```diff
@@ -178,15 +192,29 @@ func (r *components) renderComponent(ctx context.Context, acc v1alpha2.Applicati
...
+			return nil, err
+		}
 
-	if err := SetWorkloadInstanceName(traitDefs, w, c, existingWorkload); err != nil {
-		return nil, err
+		pv := fieldpath.Pave(w.UnstructuredContent())
+		if err := pv.SetString(instanceNamePath, utils.ConstructRevisionName(acc.ComponentName,
+			int64(revision))); err != nil {
+			return nil, errors.Wrapf(err, errSetValueForField, instanceNamePath, c.GetName())
+		}
```
Comment (wonderflow) [inline pkg/controller/core.oam.dev/v1alpha2/applicationconfiguration/render.go:215]: do we have a test case for this difference

**Review by wonderflow — APPROVED** [review]
LGTM after CI pass

File: pkg/controller/core.oam.dev/v1alpha2/applicationconfiguration/render.go (hunk 3 above)
Comment (ryanzhang-oss) [reply pkg/controller/core.oam.dev/v1alpha2/applicationconfiguration/render.go:215]: not yet, will add in the next PR

**Pull request merged**